weave/trait/pairing/simple_sensor_installation_state_trait.proto

```diff
@@ -0,0 +1,117 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file specifies the simple sensor installation state trait.
+ *
+ */
+
+syntax = "proto3";
+
+package weave.trait.pairing;
+
+import "wdl/wdl_options.proto";
+
+option java_outer_classname = "WeaveInternalSimpleSensorInstallationStateTrait";
+option objc_class_prefix = "SCM";
+
+/**
+ *  @brief
+ *    Simple Sensor Installation State Trait
+ *
+ *  A Weave Common service published trait that provides a means for an
+ *  application to set a device of type SENSOR into different states
+ *  during installation, and to query the current installation state of
+ *  the device.
+
+ *
+ *  The consumer of this information is any actor driving pairing flows
+ *  for SENSOR devices.
+ */
+message SimpleSensorInstallationStateTrait {
+  option (wdl.message_type) = TRAIT;
+
+  option (wdl.trait) = {
+    stability: ALPHA,
+    vendor_id: 0x0000,
+    id:        0x1402,
+    version:   1
+  };
+
+  //  @brief
+  //    Device Modes during installation
+  //
+  //  Different user interface modes used during device installation
+  //
+  enum DeviceMode {
+    // Unspecified status, possibly what the device starts with
+    DEVICE_MODE_UNSPECIFIED = 0;
+
+    // Idle: the device shows no active UX when in this state.
+    DEVICE_MODE_IDLE = 1;
+
+    // Placement guidance: the device displays visual indications, such as
+    // lighting up, which will assist the user with installation
+    DEVICE_MODE_PLACEMENT_GUIDANCE = 2;
+
+    // Self test: the device runs a self test, such as to test if it is
+    // open or closed properly
+    DEVICE_MODE_SELF_TEST = 3;
+  }
+
+  //  @brief
+  //    Self test result Status
+  //
+  //  The current result of a device self test.
+  //
+  enum SelfTestResult {
+    SELF_TEST_RESULT_UNSPECIFIED = 0;
+    SELF_TEST_RESULT_UNKNOWN = 1;
+    SELF_TEST_RESULT_INVALID = 2;
+    SELF_TEST_RESULT_VALID = 3;
+  }
+
+  //  @brief
+  //    Response to command to set device mode.
+  enum SetDeviceModeResponseType {
+    SET_DEVICE_MODE_RESPONSE_TYPE_UNSPECIFIED = 0;
+    SET_DEVICE_MODE_RESPONSE_TYPE_SUCCESS = 1;  // success
+    SET_DEVICE_MODE_RESPONSE_TYPE_ERROR = 2;  // error
+  }
+
+  // ---------- PROPERTIES ---------- //
+
+  // Which mode the device is currently on.
+  DeviceMode device_mode = 1;
+
+  SelfTestResult self_test_result = 2;
+
+  // ----------- COMMANDS ----------- //
+
+  message SetDeviceModeRequest {
+    option (wdl.message_type) = COMMAND;
+    option (wdl.command) = {
+      id: 0x01
+    };
+
+    // The device mode to switch to.
+    DeviceMode device_mode = 1;
+  }
+};
```

Comment:
Add a newline,s.t. the file ends with a newline.

Reply:
Done.